### PR TITLE
Rls 0.24.0

### PR DESCRIPTION
**Pull request by TomAugspurger**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -21,7 +21,8 @@ requirements:
     - python
     - pip
     - cython
-    - numpy
```
Comment (TomAugspurger) [inline recipe/meta.yaml]: I'm not sure about these changes. This version bumps the minimum required numpy to 1.12.
Following https://conda-forge.org/docs/meta.html#building-against-numpy, I would want something like

```
host:
  - numpy 1.12.*
run:
  - {{ pin_compatible('numpy') }}
```

But NumPy 1.12 (and 1.13) don't support Python 3.7, so we need numpy 1.14.* for py37.

Comment (TomAugspurger) [reply recipe/meta.yaml]: When running locally `conda build --python=3.7 recipe/ `

conda can't resolve the right dependencies

```
conda.exceptions.UnsatisfiableError: The following specifications were found to be in conflict:
  - numpy=1.12 -> python[version='>=2.7,<2.8.0a0']
  - python=3.7
Use "conda info <package>" to see the dependencies for each package.
```

**Comment by TomAugspurger** [comment]
Thanks to @jjhelmus, it seems like we want

```
  - numpy 1.14.*  # [py==37]
  - numpy 1.12.*  # [py!=37]
```

for now. If that sounds right, I can make a PR updating https://conda-forge.org/docs/meta.html#building-against-numpy

**Comment by jjhelmus** [comment]
See conda/conda-build#3361

**Pull request merged**